grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -86,6 +87,37 @@ class PicklistService {
         productAvailabilityService.refreshProductsAvailability(orderItem?.order?.origin?.id, [orderItem?.product?.id], binLocations?.unique(), false)
     }
 
+    // It expects to receive a picklist id
+    void clearPicklist(String id) {
+        Picklist picklist = Picklist.get(id)
+
+        if (!picklist) {
+            throw new ObjectNotFoundException(id, Picklist.class.toString())
+        }
+        // Store bin locations' id for product availability refresh
+        List<String> binLocations = []
+
+        Set<PicklistItem> picklistItems = picklist.picklistItems
+        List<PicklistItem> itemsToRemove = []
+        picklistItems.each { PicklistItem picklistItem ->
+            picklistItem.disableRefresh = true
+            itemsToRemove.add(picklistItem)
+            binLocations.add(picklistItem.binLocation?.id)
```

Comment:
won't this add a null to the list if there's no binLocation? What's going to happen in the productAvailability if that happens?

Reply:
actually we have a use case for the `null` binLocation. It's the Default bin location. It's quite difficult to understand on first sight, so maybe either Justin or Artur could explain it in more details if you were curious.

Reply:
Essentially, looking at the stockcard of a product:
![Screenshot from 2024-05-08 15-45-54](https://github.com/openboxes/openboxes/assets/93163821/3d9b0120-5bcc-4066-b5d5-1cc8e5a51878)

the first inventory item's binLocation is `null`.

Reply:
gotcha. Thanks for the explanation!